### PR TITLE
allow key-import+BORG_KEY_FILE to create key files

### PR DESCRIPTION
**Pull request by strager**
Running 'borg key import' on a keyfile repository with the BORG_KEY_FILE environment variable set works correctly if the BORG_KEY_FILE file already exists. However, the command crashes if the BORG_KEY_FILE file does not exist:

    $ BORG_KEY_FILE=newborgkey borg key import /home/strager/borg-backups/straglum borgkey
    Local Exception
    Traceback (most recent call last):
      [snip]
      File "[snip]/borg/crypto/key.py", line 713, in sanity_check
        with open(filename, 'rb') as fd:
    FileNotFoundError: [Errno 2] No such file or directory: '[snip]/newborgkey'

    Platform: Linux straglum 5.0.0-25-generic #26~18.04.1-Ubuntu SMP Thu Aug 1 13:51:02 UTC 2019 x86_64
    Linux: debian buster/sid
    Borg: 1.1.11  Python: CPython 3.7.7 msgpack: 0.5.6
    PID: 15306  CWD: /home/strager/Projects/borg
    sys.argv: ['[snip]/borg', 'key', 'import', '/home/strager/borg-backups/straglum', 'borgkey']
    SSH_ORIGINAL_COMMAND: None

Make 'borg key import' not require the BORG_KEY_FILE file to already exist.

This commit does not change the behavior of 'borg key import' without BORG_KEY_FILE. This commit also does not change the behavior of 'borg key import' on a repokey repository.

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -3815,6 +3815,15 @@ def define_borg_mount(parser):
...
         proceeding to the next line. For this format PATH must not be given.
+
+        For repositories using keyfile encryption, the key file which ``borg key
+        import`` writes to depends on several factors. If the ``BORG_KEY_FILE``
+        environment variable is set and non-empty, ``borg key import`` creates
+        or overwrites that file named by ``$BORG_KEY_FILE``. Otherwise, ``borg
+        key import`` searches in the ``$BORG_KEYS_DIR`` directory for a key file
+        associated with the repository. If an key file is found in
+        ``$BORG_KEYS_DIR``, ``borg key import`` overwrites it; otherwise, ``borg
+        key import`` creates a new key file in ``$BORG_KEYS_DIR``.
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3826]: you only need to commit this change, committing the changes to generated files is done at release time in a separate commit.

File: src/borg/archiver.py [hunk 2]
```diff
@@ -3869,6 +3869,15 @@ def define_borg_mount(parser):
         If the ``--paper`` option is given, the import will be an interactive
         process in which each line is checked for plausibility before
         proceeding to the next line. For this format PATH must not be given.
+
+        For repositories using keyfile encryption, the key file which ``borg key
+        import`` writes to depends on several factors. If the ``BORG_KEY_FILE``
+        environment variable is set and non-empty, ``borg key import`` creates
+        or overwrites that file named by ``$BORG_KEY_FILE``. Otherwise, ``borg
+        key import`` searches in the ``$BORG_KEYS_DIR`` directory for a key file
+        associated with the repository. If an key file is found in
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:3878]: If **a key** is found ...

File: docs/usage/key.rst [hunk 3]
```diff
@@ -26,6 +26,11 @@ Examples
     Remember your passphrase. Your data will be inaccessible without it.
     Key updated
 
+    # Import a previously-exported key into the specified
+    # key file (creating or overwriting the output key)
```
Comment (ThomasWaldmann) [inline docs/usage/key.rst:30]: maybe add a dot at the end of the sentence.

Comment (strager) [reply docs/usage/key.rst:30]: I'm following the convention of other comments in this documentation (e.g. lines 9 and 21) which do not have dots ending the sentences.

Comment (ThomasWaldmann) [reply docs/usage/key.rst:30]: hmm, ok, maybe this should be cleaned up later and in general.

i try to follow the rule "if it is a complete sentence, then it should have a full-stop at the end".

**Comment by ThomasWaldmann** [comment]
Thanks!

**Comment by ThomasWaldmann** [comment]
Do we need a 1.1-maint backport of this?
